aws/resource_aws_kinesis_analytics_application.go

```diff
@@ -0,0 +1,868 @@
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "ResourceNotFoundException" {
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("[WARN] Error reading Kinesis Analytics Application: \"%s\", code: \"%s\"", awsErr.Message(), awsErr.Code())
+		}
+		return err
+	}
+
+	d.SetId(aws.StringValue(resp.ApplicationDetail.ApplicationARN))
+	d.Set("name", aws.StringValue(resp.ApplicationDetail.ApplicationName))
+	d.Set("arn", aws.StringValue(resp.ApplicationDetail.ApplicationARN))
+	d.Set("code", aws.StringValue(resp.ApplicationDetail.ApplicationCode))
+	d.Set("create_timestamp", aws.TimeValue(resp.ApplicationDetail.CreateTimestamp).Format(time.RFC3339))
+	d.Set("description", aws.StringValue(resp.ApplicationDetail.ApplicationDescription))
+	d.Set("last_update_timestamp", aws.TimeValue(resp.ApplicationDetail.LastUpdateTimestamp).Format(time.RFC3339))
+	d.Set("status", aws.StringValue(resp.ApplicationDetail.ApplicationStatus))
+	d.Set("version", int(aws.Int64Value(resp.ApplicationDetail.ApplicationVersionId)))
+
+	if err := d.Set("cloudwatch_logging_options", getCloudwatchLoggingOptions(resp.ApplicationDetail.CloudWatchLoggingOptionDescriptions)); err != nil {
+		return err
+	}
+
+	if err := d.Set("inputs", getInputs(resp.ApplicationDetail.InputDescriptions)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func resourceAwsKinesisAnalyticsApplicationUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).kinesisanalyticsconn
+
+	if !d.IsNewResource() {
+		applicationUpdate := &kinesisanalytics.ApplicationUpdate{}
+		name := d.Get("name").(string)
+		version := d.Get("version").(int)
+
+		updateApplicationOpts := &kinesisanalytics.UpdateApplicationInput{
+			ApplicationName:             aws.String(name),
+			CurrentApplicationVersionId: aws.Int64(int64(version)),
+		}
+
+		applicationUpdate, err := createApplicationUpdateOpts(d)
+		if err != nil {
+			return err
+		}
+
+		if !reflect.DeepEqual(applicationUpdate, &kinesisanalytics.ApplicationUpdate{}) {
+			updateApplicationOpts.SetApplicationUpdate(applicationUpdate)
+			_, updateErr := conn.UpdateApplication(updateApplicationOpts)
+			if updateErr != nil {
+				return updateErr
+			}
+			version = version + 1
+		}
+
+		oldLoggingOptions, newLoggingOptions := d.GetChange("cloudwatch_logging_options")
+		if len(oldLoggingOptions.([]interface{})) == 0 && len(newLoggingOptions.([]interface{})) > 0 {
+			if v, ok := d.GetOk("cloudwatch_logging_options"); ok {
+				clo := v.([]interface{})[0].(map[string]interface{})
+				cloudwatchLoggingOption := createCloudwatchLoggingOption(clo)
+				addOpts := &kinesisanalytics.AddApplicationCloudWatchLoggingOptionInput{
+					ApplicationName:             aws.String(name),
+					CurrentApplicationVersionId: aws.Int64(int64(version)),
+					CloudWatchLoggingOption:     cloudwatchLoggingOption,
+				}
+				conn.AddApplicationCloudWatchLoggingOption(addOpts)
```

Comment:
Not really sure the best way to handle this but adding new cloudwatch logging options cannot be part of the `UpdateApplication` call.